File: rust/arDEWino-rs/src/calculations.rs
pub fn calculate_dewpoint(t: f32, rh: f32) -> f32 {
    if !(0.0..=100.0).contains(&rh) {
        panic!("Relative humidity must be between 0 and 100.");
    }
    let rh_decimal = rh / 100.0;
    let log_rh = rh_decimal.ln();
    let numerator = 243.04 * (log_rh + ((17.625 * t) / (243.04 + t)));
    let denominator = 17.625 - log_rh - ((17.625 * t) / (243.04 + t));
    numerator / denominator
}

pub fn round_to_2_decimal_places(value: f32) -> f32 {
    (value * 100.0).round() / 100.0
}

File: go/dewdrop-go/pkg/calculations/calculations.go
package calculations

import (
	"errors"
	"math"
)

// DewPointCalculator uses Magnus-Tetens formula for dew point
func DewPointCalculator(temperature, relativeHumidity float64) (float64, error) {
	if temperature < -273.15 {
		return temperature, errors.New("temperature must be greater than or equal to -273.15")
	}
	if relativeHumidity < 0 || relativeHumidity > 100 {
		return temperature, errors.New("relative humidity must be between 0 and 100")
	}
	if math.IsNaN(temperature) || math.IsNaN(relativeHumidity) {
		return temperature, errors.New("input values must not be NaN")
	}
	if math.IsInf(temperature, 0) || math.IsInf(relativeHumidity, 0) {
		return temperature, errors.New("input values must not be infinite")
	}

	t := temperature
	rh := relativeHumidity
	dewPoint := (243.04 * (math.Log(rh/100) + ((17.625 * t) / (243.04 + t)))) /
		(17.625 - math.Log(rh/100) - ((17.625 * t) / (243.04 + t)))

	if math.IsNaN(dewPoint) {
		return 0, errors.New("result is NaN")
	}
	return dewPoint, nil
}

func RoundTo2DecimalPlaces(value float32) float32 {
	return float32(math.Round(float64(value)*100.0)) / 100.0
}
